### PR TITLE
PermissionManager should ignore intermittent errors

### LINKED ISSUE #6492: PermissionManager.getPermissions() error after connection reset
I am continue with this issue since i don't have solution
[(https://github.com/realm/realm-java/issues/6458)]

After connection reset or dropped connection `PermissionManager.getPermissions()` return this error:

` E/REALM_SYNC: Connection[4]: Reading failed: Connection reset by peer
E/REALM_JAVA: Error in __permission:
    CONNECTION_RESET_BY_PEER(realm.basic_system:104): Connection reset by peer
E/REALM_SYNC: Connection[1]: Reading failed: Connection reset by peer
 E/REALM_JAVA: Error in __wildcardpermissions:
    CONNECTION_RESET_BY_PEER(realm.basic_system:104): Connection reset by peer
 E/REALM_SYNC: Connection[2]: Reading failed: Connection reset by peer`

Even if it network error like @cmelchior said, i think something wrong here.
If you just disable INTERNET manually for a second getPermissions() not working anymore even with new `PermissionManager` instance.

as i know `__permission` is a simple full-sync realm, and it should to work like other user's `Realms`, and user's Realms work fine after connection reset.

**Goals**
Call `PermissionManager.getPermissions()` to get the list of permissions for all Realms available to this user

**Expected results**
`PermissionManager.getPermissions()` should return `RealmResults<Permission>`

**Actual results**
`PermissionManager.getPermissions()` triggers `onError()` with `null` object but with some addiotinal logs from  `E/REALM_SYNC`

**Steps to reproduce**
Call `PermissionManager.getPermissions()` then manually disable and enable network then call `PermissionManager.getPermissions()` again, only after activity restart(setting configuration and connecting to realm) the `PermissionManager.getPermissions()` work again.
I want to mention that some connection reset happen by itself after about 5 minutes when `PermissionManager.getPermissions()` is called, this is happen also in the full sync TODO app example
[todo app](https://github.com/realm/my-first-realm-app/tree/master/android/todo-full-sync)

Also does `Permission`/`PermissionManager` not support offline mode?

Realm 5.10.0
Realm platform cloud - standard plan
Android studio 3.4
Checked API 25/26

### PR DESCRIPTION
**Pull request by cmelchior**
Closes #6492
Adds functionality for partly working around https://github.com/realm/realm-java-private/issues/259

Before this PR, the `PermissionManager` held on to any error reported, even intermittent network errors. This caused it to stop working once such an error was encountered. 

With this PR, all errors known to be temporary are just ignored, since the end user cannot do anything besides retrying closing/opening the PermissionManager again. Internally we are already retrying sending data which is much more efficient, so no reason to put that burden on the end user.

With this change, only non-recoverable errors should be reported to end users.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,20 @@
+## 5.12.0(YYYY-MM-DD)
+
+### Enhancements
+* [ObjectServer] Added `SyncManager.refreshConnections()` that can be used to manually trigger a reconnect for all sessions. This is useful if the device has been offline for a long time or fail to detect that the device regained connectivity. (Issue [#259](https://github.com/realm/realm-java-private/issues/259))
```
Comment (kneth) [inline CHANGELOG.md]:
```suggestion
* [ObjectServer] Added `SyncManager.refreshConnections()` that can be used to manually trigger a reconnect for all sessions. This is useful if the device has been offline for a long time or fail to detect that it regained connectivity. (Issue [#259](https://github.com/realm/realm-java-private/issues/259))
```

File: realm/realm-library/src/syncIntegrationTest/java/io/realm/PermissionManagerTests.java [hunk 2]
```diff
@@ -506,7 +506,7 @@ public void permissionManagerAsyncTask_handlePermissionRealmError() throws NoSuc
         // Simulate error in the permission Realm
         Field permissionConfigField = pm.getClass().getDeclaredField("permissionRealmError");
         permissionConfigField.setAccessible(true);
-        final ObjectServerError error = new ObjectServerError(ErrorCode.UNKNOWN, "Boom");
+        final ObjectServerError error = new ObjectServerError(ErrorCode.WRONG_PROTOCOL_VERSION, "Boom");
```
Comment (bmunkholm) [inline realm/realm-library/src/syncIntegrationTest/java/io/realm/PermissionManagerTests.java]: "Boom"?? What does that mean :-)

Comment (bmunkholm) [reply realm/realm-library/src/syncIntegrationTest/java/io/realm/PermissionManagerTests.java]: Same for all the other "Boom"'s - is that propagated to the user? Then change to a better description.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ## 5.12.0(YYYY-MM-DD)
 
 ### Enhancements
-* [ObjectServer] Added `SyncManager.refreshConnections()` that can be used to manually trigger a reconnect for all sessions. This is useful if the device has been offline for a long time or fail to detect that the device regained connectivity. (Issue [#259](https://github.com/realm/realm-java-private/issues/259))
+* [ObjectServer] Added `SyncManager.refreshConnections()` that can be used to manually trigger a reconnect for all sessions. This is useful if the device has been offline for a long time or fail to detect that it regained connectivity. (Issue [#259](https://github.com/realm/realm-java-private/issues/259))
 
 ### Fixed
 * [ObjectServer] `PermissionManager` stopped working if an intermittent network error was reported. (Issue [#6492](https://github.com/realm/realm-java/issues/6492))
```
Comment (bmunkholm) [inline CHANGELOG.md]: "Since" ?

Comment (Zhuinden) [reply CHANGELOG.md]: I don't think PermissionManager has had many changes made to its error handling since its introduction. So that would mean 3.7.0.

**Pull request merged**